devito/nodes.py

```diff
@@ -473,9 +474,19 @@ def __init__(self, name, body, retval, parameters=None, prefix=('static', 'inlin
         self.name = name
         self.body = as_tuple(body)
         self.retval = retval
-        self.parameters = as_tuple(parameters)
         self.prefix = prefix
 
+        if all([isinstance(x, RuntimeArgProvider) for x in parameters]):
+            args = flatten([x.rtargs for x in parameters])
+        else:
+            assert(all([isinstance(x, RuntimeArgument) for x in parameters]))
+            args = parameters
+
+        self.parameters = as_tuple(args)
+
+        self.t_args = [x for x in args if isinstance(x, TensorArgument)]
```

Comment:
@FabioLuporini You haven't continued the discussion here. I am not convinced that it is wrong to not use getters here.